libwallet/src/api_impl/owner_swap.rs

```diff
@@ -636,6 +735,12 @@ where
 	C: NodeClient + 'a,
 	K: Keychain + 'a,
 {
+	let ethereum_wallet = wallet_inst
+		.clone()
+		.lock()
+		.lc_provider()?
+		.get_ethereum_wallet()?;
+
```

Comment:
Can you do get_ethereum_wallet  from w  below?  I think wallet_lock! does the same